utils/i18nUtils.js

```diff
@@ -0,0 +1,97 @@
+import i18next from "i18next";
+import { initReactI18next } from "react-i18next";
+import resourcesToBackend from "i18next-resources-to-backend";
+import { match } from "@formatjs/intl-localematcher";
+import { shouldPolyfill } from "@formatjs/intl-locale/should-polyfill";
+import {
+  DEFAULT_LOCALE as DEFAULT_POLARIS_LOCALE,
+  SUPPORTED_LOCALES as SUPPORTED_POLARIS_LOCALES,
+} from "@shopify/polaris";
+
+// Update this to the default locale for the app
+const DEFAULT_APP_LOCALE = "en";
+// Update this to the supported locales contained in the `locales` folder
+const SUPPORTED_APP_LOCALES = ["en", "fr"];
+
+let _userLocale, _polarisTranslations;
+
+export function getUserLocale() {
+  if (_userLocale) {
+    return _userLocale;
+  }
+  const url = new URL(window.location.href);
+  const locale = url.searchParams.get("locale") || DEFAULT_APP_LOCALE;
+  _userLocale = match([locale], SUPPORTED_APP_LOCALES, DEFAULT_APP_LOCALE);
+  return _userLocale;
+}
+
+export function getPolarisTranslations() {
+  return _polarisTranslations;
+}
+
+export async function initI18n() {
+  if (shouldPolyfill()) {
+    await import("@formatjs/intl-locale/polyfill");
+  }
+  await Promise.all([initI18next(), fetchPolarisTranslations()]);
+}
+
+async function initI18next() {
+  return await i18next
+    .use(initReactI18next)
+    .use(localResourcesToBackend())
+    .init({
+      debug: process.env.NODE_ENV === "development",
+      lng: getUserLocale(),
+      fallbackLng: DEFAULT_APP_LOCALE,
+      supportedLngs: SUPPORTED_APP_LOCALES,
+      interpolation: {
+        // React escapes values by default
+        escapeValue: false,
+      },
+      react: {
+        // Wait for the locales to be loaded before rendering the app
+        // instead of using a Suspense component
+        useSuspense: false,
+      },
+    });
+}
+
+function localResourcesToBackend() {
+  return resourcesToBackend(async (locale, _namespace, callback) => {
+    import(`../locales/${locale}.json`)
```

Comment:
App extensions and themes use the `.default.json` extension to indicate the default locale translation file. I contend that we should follow that pattern in this template too.

`en.json` -> `en.default.json`

Benefits:
 * consistency through all parts of the app
 * We'll be able to auto-configure `DEFAULT_APP_LOCALE` and `SUPPORTED_APP_LOCALES` so that developers don't need to set those themselves
 * People can learn how to create a different default locale by seeing how this code work

Something like:

```suggestion
    import(`../locales/${locale}${locale === i18next.fallbackLng ? '.default' : ''}.json`)
```